all-actors/src/main/java/org/sunbird/cert/actor/CertificateGeneratorActor.java

```diff
@@ -103,7 +104,24 @@ private void generateSignUrl(Request request) {
 
 
     private BaseStorageService getStorageService() {
-        StorageConfig storageConfig = new StorageConfig(certVar.getCloudStorageType(), certVar.getCloudStorageKey(), certVar.getCloudStorageSecret());
+        StorageConfig storageConfig;
+        String storageKey = certVar.getCloudStorageKey();;
+        String storageSecret = certVar.getCloudStorageSecret();
+        Option<String> storageEndpoint;
+        Option<String> storageRegion;
+        if (certVar.getCloudStorageType().equalsIgnoreCase("azure")) {
+            storageEndpoint = Option.apply("");
+            storageRegion = Option.apply("");
+        } else if (certVar.getCloudStorageType().equalsIgnoreCase("oci")){
+            storageEndpoint = Option.apply(certVar.getCloudStorageEndpoint());
+            storageRegion = Option.apply("");
+        } else {
+            storageKey = "";
+            storageSecret = "";
+            storageEndpoint = scala.Option.apply("");
+            storageRegion = scala.Option.apply("");
+        }
```

Comment:
Instead of storage type check can we follow the same as done here https://github.com/Sunbird-Lern/sunbird-lms-service/pull/1163/files

Reply:
it may work, but need additional testing. This one is known to work. There is scope of improvement but i suggest it take it in the future releases

Reply:
removed unwanted switch